packages/cli/src/metadataGeneration/metadataGenerator.ts

```diff
@@ -111,6 +112,126 @@ export class MetadataGenerator {
+        if (
+          (tester.paths[i].isParam && !subject.paths[i].isParam) ||
+          (!tester.paths[i].isParam && subject.paths[i].isParam) ||
+          (!tester.paths[i].isParam && tester.paths[i].path !== subject.paths[i].path)
+        ) {
+          return _PathDuplicationType.NONE;
+        }
+      }
+      if (tester.paths.length === subject.paths.length) {
+        return _PathDuplicationType.FULL;
+      } else if (tester.paths.length > subject.paths.length) {
+        return _PathDuplicationType.PARTIAL_SUBJECT_PATH_IS_TESTER_PREFIX;
+      } else {
+        return _PathDuplicationType.PARTIAL_TESTER_PATH_IS_SUBJECT_PREFIX;
+      }
+    }
+
+    controllers.forEach(controller => {
+      const methodRouteGroup: {
+        [key: string]: Array<{
+          paths: Array<{
+            isParam: boolean;
+            path: string;
+          }>;
+          method: Tsoa.Method;
+        }>;
+      } = {};
+      // Group all ts methods with HTTP method decorator into same object in same controller.
+      controller.methods.forEach(method => {
+        if (methodRouteGroup[method.method] === undefined) {
+          methodRouteGroup[method.method] = [];
+        }
+        methodRouteGroup[method.method].push({
+          paths: method.path.split('/').map((val: string) => {
+            return { isParam: isParamRegExp.test(val), path: val };
+          }),
+          method,
+        });
+      });
+
+      const dupRoute: { [key: string]: Tsoa.Method[] } = {};
+      Object.keys(methodRouteGroup).forEach((key: string) => {
+        const methodRoutes: Array<{ paths: Array<{ isParam: boolean; path: string }>; method: Tsoa.Method }> = methodRouteGroup[key];
+        const duplicates: Tsoa.Method[] = [];
+        for (let i = 0; i < methodRoutes.length - 1; i += 1) {
+          const iMethodRoute = methodRoutes[i];
+          for (let j = i + 1; j < methodRoutes.length; j += 1) {
+            const jMethodRoute = methodRoutes[j];
+            const examineResult = _examinePaths(iMethodRoute, jMethodRoute);
+            if (examineResult === 1) {
+              if (!duplicates.includes(iMethodRoute.method)) {
+                duplicates.push(iMethodRoute.method);
+              }
+              if (!duplicates.includes(jMethodRoute.method)) {
+                duplicates.push(jMethodRoute.method);
+              }
+            } else if (examineResult === 2) {
+              console.warn(
+                `[Method ${jMethodRoute.method.name} route: ${jMethodRoute.method.path}] may never been invoke, because its route is partialy collison with [Method ${iMethodRoute.method.name} route: ${iMethodRoute.method.path}]`,
+              );
+            } else if (examineResult === 3) {
+              console.warn(
+                `[Method ${iMethodRoute.method.name} route: ${iMethodRoute.method.path}] may never been invoke, because its route is partialy collison with [Method ${jMethodRoute.method.name} route: ${jMethodRoute.method.path}]`,
```

Comment:
```suggestion
                `[Method ${iMethodRoute.method.name} route: ${iMethodRoute.method.path}] may never be invoked, because its route partially collides with [Method ${jMethodRoute.method.name} route: ${jMethodRoute.method.path}]`,
```